content/blogposts/2022/code-intelligence-platform.md

```diff
@@ -0,0 +1,108 @@
+
+For our customers, Sourcegraph is the one universal dev tool: the only tool used by all the devs and the only tool that indexes *all* of the code. Other tools in the dev workflow are fragmented: devs use different editors and languages, and often there are multiple code hosts.
+
+Devs want a single place with all the information about code: code navigation (definitions and references), authorship, ownership, dependency graph, code scans, deployment status, observability (whether a line of code is throwing lots of errors), etc.
+
+Devs want Sourcegraph to be this place, and to help them find answers instead of just the raw information. We often hear things like:
+
+- "I'm trying to figure out the best way to call a function. Don't just show me a list of call sites. Show me how the code owners of that function themselves have recently been calling that function."
+- "I need to debug an issue in code. Don't just let me navigate within the same project to understand it. Let me see what changed and why, and trace calls across projects and dependencies."
+- "I need to see how my team's API is being used. Don't just show me a list of file locations. Show me which other teams are using it and the trend of usage over time, grouped by usage pattern."
+- "I need to fix something across my entire codebase? Don't just help me find the spots that need fixing. Let me push up a quick fix to all affected repositories, tracks its progress, and monitor for regressions."
+
+Code search isn't enough to answer these kinds of questions, because code search doesn't know all the other necessary information about code. Code search just helps you find matching lines in code. TODO sqs - (thats why we are calling it ...)
+
+We added [Code Insights](https://docs.sourcegraph.com/code_insights) to let devs track trends over time, [Batch Changes](https://docs.sourcegraph.com/batch_changes) to let devs make large-scale code changes across many repositories, and more. These crucial primitives helped, but we still needed to teach Sourcegraph much more about code and its associated metadata.
+
+Remember how Google went from helping you find things *on* the web (matching web pages) to helping you learn things *from* the web (information and answers) by connecting data like news, videos, and maps back to web search? Today, you go to Google to answer complex questions and learn things, not to find things.
```

Comment:
```suggestion
Remember how Google went from helping you find things *on* the web (matching web pages) to helping you learn things *from* the web (information and answers) by connecting data like news, videos, and maps back to web search? Today, you go to Google to answer complex questions and learn things, not to find things.

We've spent 9 years building the underlying foundation of code search to be fast, secure, and scalable across ~10^11 lines of code for the world's most demanding software teams, and we're hard at work making Sourcegraph even smarter. 

Today, we're releasing [Sourcegraph 4.0](/blog/release/4.0). Sourcegraph 4.0 is the first step toward Sourcegraph becoming a code intelligence platform, and is available today. The release includes over a dozen updates and new features that make answering questions *about* code as easy as searching for matching lines *in* code. 

Today, we're also launching [Sourcegraph Cloud:](/blog/enterprise-cloud) secure, scalable, dedicated Sourcegraph instances on the cloud. Sourcegraph Cloud is the best way to start using Sourcegraph on your organization's code. [Sign up now](https://signup.sourcegraph.com/).
```

Alternative order for the conclusion

Reply:
The post would end here if you go this route.

Reply:
I'd delete the "Today," in the last paragraph here if this is the way things go.

Reply:
thx